comp/metadata/inventoryagent/inventoryagentimpl/inventoryagent.go

```diff
@@ -293,6 +296,9 @@ func (ia *inventoryagent) getPayload() marshaler.JSONMarshaler {
 
 // Get returns a copy of the agent metadata. Useful to be incorporated in the status page.
 func (ia *inventoryagent) Get() map[string]interface{} {
+	ia.m.Lock()
```

Comment:
Note: Did you consider moving the lock to the caller, ie, where there are several goroutines?

Reply:
I think it's better to have the inventory component a black box offering a set of feature to others rather than having all the users lock before calling a function.